app/celery/scheduled_tasks.py

```diff
@@ -129,6 +130,17 @@ def switch_current_sms_provider_on_slow_delivery():
                 dao_reduce_sms_provider_priority(provider_name, time_threshold=timedelta(minutes=10))
 
 
+@notify_celery.task(name="generate-sms-delivery-stats")
+def generate_sms_delivery_stats():
+    for delivery_interval in (1, 5, 10):
+        providers_slow_delivery_ratios = get_ratio_of_messages_delivered_slowly_per_provider(
+            created_within_minutes=10, delivered_within_minutes=delivery_interval
```

Comment:
I wonder, would it affect stats that we are looking for notifications that took longer than 10 minutes to deliver within a set of notifications that were created within last 10 minutes only?

Reply:
Oo great question Pea!

So I think you've spotted an issue there.

If `created_after` is 10 minutes ago, it means we only include notifications that were created less than or equal to 10 minutes ago.

There can't be a notification that is both
- created less than or equal to 10 minutes ago
- sent at equal to or older than 10 minutes ago

The only exception is a notification that is exactly 10 minutes old and still in sending.

I think there are two options here

1. Change the time period to consider notifications sent in say the last 15 minutes. For consistency, I'd suggest we do for all of the stats (1 min sending times, 5 min sending times, 10 min sending times)
2. Remove the 10 minute stat and just keep the 1 min and 5 min stat

I'm going to suggest we do option 1 for these.

What do you think Pea?

Reply:
I also think suggestion 1 works best ☺️

Reply:
Thanks Pea, I've added a commit on the end that implements option 1. Let me know what you think please!